### PR TITLE
Drop Python 3.7 support

### PR DESCRIPTION
**Pull request by zerolab**
Python 3.7 is EOL since 2 weeks ago. https://endoflife.date/python 
This PR drops support for it, and runs pyupgrade with a target of 3.8+ (3ed5688)

Also runs django-upgrade with a 3.2 target

Note that Willow 1.6 dropped support for Python 3.7 and is the remaining failure on #10657.

### CODE REVIEW

**Comment by lb-** [comment]
Just checking, do any of the translation changes go against our documented conventions?
https://docs.wagtail.org/en/stable/contributing/translations.html#marking-strings-for-translation

> In both Python and templates, make sure to always use named placeholder. In addition, in Python, only use the printf style formatting. This is to ensure compatibility with Transifex and help translators in their work.

**Comment by zerolab** [comment]
@lb- as far as I can see none of the string changes are for translated strings, so they should not be going against our conventions

File: scripts/get-translator-credits.py [hunk 1]
```diff
@@ -60,7 +59,7 @@ def get_language_name(locale_string):
 language_names.sort()
 
 for (language_name, locale) in language_names:
-    print(("%s - %s" % (language_name, locale)))  # noqa: T201
+    print("{} - {}".format(language_name, locale))  # noqa: T201
```
Comment (laymonage) [inline scripts/get-translator-credits.py:62]: Curious: this changes `%`-formatting to `.format()`, but in other places we have `.format()` being replaced with `f`-strings. Does this mean re-running pyupgrade will change this to use f-string?

Comment (zerolab) [reply scripts/get-translator-credits.py:62]: @laymonage ha! didn't occur to me to try again. It does. Shall I commit and push?

Comment (laymonage) [reply scripts/get-translator-credits.py:62]: Would be nice to be consistent, but I don't know if @gasman has started reviewing or not

File: wagtail/embeds/finders/instagram.py [hunk 2]
```diff
@@ -66,7 +66,7 @@ def find_embed(self, url, max_width=None, max_height=None):
 
         # Convert photos into HTML
         if oembed["type"] == "photo":
-            html = '<img src="%s" alt="">' % (oembed["url"],)
+            html = '<img src="{}" alt="">'.format(oembed["url"])
```
Comment (RealOrangeOne) [inline wagtail/embeds/finders/instagram.py:69]: Suggestion: `format_html`

File: scripts/get-translator-credits.py (hunk 1 above)
Comment (zerolab) [reply scripts/get-translator-credits.py]: pushed

File: wagtail/images/models.py [hunk 3]
```diff
@@ -174,7 +174,7 @@ def open_file(self):
                     image_file = storage.open(self.file.name, "rb")
 
                 close_file = True
-        except IOError as e:
+        except OSError as e:
```
Comment (RealOrangeOne) [inline wagtail/images/models.py:177]: Ditto RE external storage

**Review by RealOrangeOne — COMMENTED** [review]
Lots of my comments ended up being about f-strings :see_no_evil: I got half way and stopped, but if we're going through the effort of switching to f-strings, it's probably worth doing it properly.

A couple are definitely not about f-strings though, honest!

File: wagtail/utils/widgets.py [hunk 4]
```diff
@@ -29,7 +29,7 @@ def render(self, name, value, attrs=None, renderer=None):
         widget_html = self.render_html(name, value_data, attrs)
 
         js = self.render_js_init(id_, name, value_data)
-        out = "{0}<script>{1}</script>".format(widget_html, js)
+        out = f"{widget_html}<script>{js}</script>"
```
Comment (RealOrangeOne) [inline wagtail/utils/widgets.py:32]: Suggestion: `format_html`

File: scripts/get-translator-credits.py (hunk 1 above)
Comment (gasman) [reply scripts/get-translator-credits.py]: I had, but I heartily approve of this update - having to context-switch between the different formatting variants was making the reviewing slower than it needed to be :-)

**Comment by gasman** [comment]
I suggest we address the `format_html` suggestions in a follow-up PR - while they're definitely worth fixing, they're not related to the 3.7 upgrade, and doing a load of case-by-case "was there actually a good reason for the original developer doing it this way" analysis (with security implications if we get it wrong) is a very different task to eyeballing a list of cosmetic changes made by an automated tool.

**Comment by gasman** [comment]
A couple of places in the docs that need updating to reflect dropping 3.7 support:

* the table on releases/upgrading.md
* the "Install dependencies" section on getting_started/tutorial.md

File: docs/getting_started/tutorial.md [hunk 5]
```diff
@@ -12,7 +12,7 @@ If you want to add Wagtail to an existing Django project instead, see [](integra
 
 ### Install dependencies
 
-Wagtail supports Python 3.7, 3.8, 3.9, 3.10, and 3.11.
+Wagtail supports Python 3.8, 3.9, 3.10, and 3.11.
```
Comment (gasman) [inline docs/getting_started/tutorial.md:15]: There's another reference to 3.7 two paragraphs below.

File: wagtail/admin/forms/account.py [hunk 6]
```diff
@@ -125,7 +125,7 @@ def save(self, commit=True):
             # will clear the now-updated field on self.instance too
             try:
                 self._original_avatar.storage.delete(self._original_avatar.name)
-            except IOError:
+            except OSError:
```
[the remark this answers is not in the document]
Comment (gasman) [reply wagtail/admin/forms/account.py:128]: This was a new one on me, but according to https://docs.python.org/3/library/exceptions.html#IOError:

> The following exceptions are kept for compatibility with previous versions; starting from Python 3.3, they are aliases of [OSError](https://docs.python.org/3/library/exceptions.html#OSError).

**Review by gasman — APPROVED** [review]
LGTM (other than the remaining reference to 3.7 in the tutorial intro)! There are still a few weird places where it's chosen to use `format` when an f-string would be clearly better, but again I'd rather leave those as they are than let the human factor muck things up :-)

(incidentally, wagtail/users/view/users.py gives a hint to what the `format` vs f-string logic might be - either it's not catching `format` expressions spanning multiple lines, or it's refusing to use f-strings if the string with placeholders exceeds a certain length)

**Comment by gasman** [comment]
Merged in aca754aa366b855e2b1e92923e8e4636808fa183 - thanks @zerolab!

**Pull request closed without merging**